docs/UserGuide.md

```diff
@@ -196,22 +196,27 @@ Clears all entries from NewAddressBook.
 
 Format: `clear`
 
-### Filter students by other fields: `filter`
-Filters students by either:
+### Filter students/teachers by other fields: `filterStudent` / `filterTeacher`
+Filters students/teachers by either:
 - tag
 - involvement
-- form class
-- medical history
 
-Format: `filter ft/FILTER CATERGORY [ft/FILTER CATERGORY]...`
+Format: 
+- `filterStudent INVOLVEMENT_FILTER_CATEGORY [t/] [TAG_FILTER CATEGORY]...`
+- `filterTeacher INVOLVEMENT_FILTER_CATEGORY [t/] [TAG_FILTER CATEGORY]...`
 
-Search is not case sensitive e.g. “student” same as “STUDENT”
+Filters by involvement first, to filter by tag, add `t/`, followed by tag terms behind
 
-Only full words will be matched e.g. “Class A1” will not match “Class A”.
+The filter category is not case-sensitive e.g. “student” same as “STUDENT” but "t/" is not the same as "T/"
 
-More than 1 filter is allowed e.g.  `filter ft/[filter category 1] ft/[filter category 2]`.
+More than 1 filter is allowed e.g.  `INVOLVEMENT_FILTER_CATERGORY [t/] [TAG_FILTER CATERGORY] [TAG_FILTER CATERGORY]`.
+
```

Comment:
Maybe it's good here to say that whether filtering by tags/involvement will match by word. ie. filter `t/so` will not match a person with tag `something`

Reply:
Currently it uses contains though, cause like if you want to see all your classes can just type `class`, then if you want to find math class in particular then search `math`.

Or is searching by exact keywords better?